mage_integrations/mage_integrations/sources/paystack/streams/base.py

```diff
@@ -0,0 +1,172 @@
+            metadata = {}
+
+            if self.catalog.metadata is not None:
+                metadata = singer.metadata.to_map(self.catalog.metadata)
+
+            return tx.transform(
+                record,
+                self.catalog.schema.to_dict(),
+                metadata
+            )
+
+    def get_stream_data(self, data):
+        return [self.transform_record(item) for item in data]
+
+    def load_data(self):
+        table = self.TABLE
+        api_method = self.API_METHOD
+        done = False
+        sync_interval_in_mins = 2
+
+        # Attempt to get the bookmark date from the state file (if one exists and is supplied).
+        LOGGER.info('Attempting to get the most recent bookmark_date for entity {}.'.format(self.ENTITY))
+        bookmark_date = get_last_record_value_for_table(self.state, table, 'bookmark_date')
+
+        # If there is no bookmark date, fall back to using the start date from the config file.
+        if bookmark_date is None:
+            LOGGER.info('Could not locate bookmark_date from STATE file. Falling back to start_date from config.json instead.')
+            bookmark_date = parse(self.config.get('start_date'))
+        else:
+            bookmark_date = parse(bookmark_date)
+
+        # Convert bookmarked start date to POSIX.
+        bookmark_date_iso = bookmark_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+        to_date = datetime.now(pytz.utc) - timedelta(minutes=sync_interval_in_mins)
+        to_date_iso = to_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+        sync_window = str([bookmark_date_iso, to_date_iso])
+        LOGGER.info("Sync Window {} for schema {}".format(sync_window, table))
+
+        params = {'from': bookmark_date_iso, 'to': to_date_iso}
+
+        LOGGER.info("Querying {} starting at {}".format(table, bookmark_date))
+
+        while not done:
+            max_date = to_date
+
+            response = self.client.make_request(
+                url=self.get_url(),
+                method=api_method,
+                params=params
+            )
+
+            meta = response.get('meta', dict())
+
+            if 'api_error_code' in response.keys():
+                if response['api_error_code'] == 'configuration_incompatible':
+                    LOGGER.error('{} is not configured'.format(response['error_code']))
+                    break
+
+            records = response.get('data')
+
+            # Get records from API response and transform
+            to_write = self.get_stream_data(records)
+
+            with singer.metrics.record_counter(endpoint=table) as ctr:
+                yield to_write
+
+                ctr.increment(amount=len(to_write))
+
+            # update max_date with minimum of (max_replication_key) or (now - 2 minutes)
+            # this will make sure that bookmark does not go beyond (now - 2 minutes)
+            # so, no data will be missed due to API latency
+            max_date = min(max_date, to_date)
+            self.state = incorporate(
```

Comment:
The bookmark property is "createdAt" in your TransactionsStream. But you use bookmark_date here, they have conflicts. 
The logic of updating state and bookmark values is here https://github.com/mage-ai/mage-ai/blob/master/mage_integrations/mage_integrations/sources/base.py#L432-L444